include/dsn/internal/rpc_message.h

```diff
@@ -150,7 +150,8 @@ namespace dsn
     public:
         message_header         *header;
         std::vector<blob>      buffers; // header included for *send* message, 
-                                        // header not included for *recieved* 
+                                        // header not included for *recieved*
+        std::vector<blob>      buffers_for_parser;
```

Comment:
This is the parser's problem, which introduces memory and performance overhead where normal rDSN msgs should not have. I prefer the original approach.